File: action-type/15-chain-of-responsibility/go/personal_info_filter.go
package main

import (
	"fmt"
	"regexp"
)

// PersonalInfoFilter 个人信息过滤器

type PersonalInfoFilter struct{}

func (f *PersonalInfoFilter) Filter(text string, chain *FilterChain) string {

	// 手机号简单替换
	re := regexp.MustCompile(`\d{11}`)
	result := re.ReplaceAllString(text, "[手机号]")
	fmt.Println("[PersonalInfoFilter] 处理后:", result)

	return chain.Filter(result)
}

File: action-type/15-chain-of-responsibility/java/PersonalInfoFilter.java
package main;

// 个人信息过滤器
public class PersonalInfoFilter implements SensitiveFilter {
    
    @Override
    public String doFilter(String text, FilterChain chain) {
    
        String result = text.replaceAll("[0-9]{11}", "[手机号]");
        System.out.println("[PersonalInfoFilter] 处理后: " + result);
    
        return chain.doFilter(result);
    }
}
